http_client.rst

```diff
@@ -891,20 +891,20 @@ Handling Exceptions
 ~~~~~~~~~~~~~~~~~~~
 
 When the HTTP status code of the response is in the 300-599 range (i.e. 3xx,
-4xx or 5xx) your code is expected to handle it. If you don't do that, the
-``getHeaders()``, ``getContent()`` and ``toArray()`` methods throw an appropriate exception, all of
-which implement the :class:`Symfony\\Contracts\\HttpClient\\Exception\\HttpExceptionInterface`::
+4xx or 5xx), the ``getHeaders()``, ``getContent()`` and ``toArray()`` methods
+throw an appropriate exception, all of which implement the
+:class:`Symfony\\Contracts\\HttpClient\\Exception\\HttpExceptionInterface`.
 
-    // the response of this request will be a 403 HTTP error
-    $response = $client->request('GET', 'https://httpbin.org/status/403');
+You can prevent the exception from being thrown by either:
 
-    // this code results in a Symfony\Component\HttpClient\Exception\ClientException
-    // because it doesn't check the status code of the response
-    $content = $response->getContent();
+* Checking for the exact status code::
+    
+    $response = $client->request('GET', 'https://httpbin.org/status/403');
+    if (403 === $response->getStatusCode()) {
+        $response->getContent();
```

Comment:
This will throw.
Checking the status code disables throwing *by the destructor* only.
The behavior of the other methods is always predictable and controlled by the `$throw` argument.
I agree we should find a better way to explain all this :)

Reply:
Damn, I checked for *string* `'403'`, while in fact it's `int`...
Please take a look again!

And if you still have some nerves left: :-)
Maybe it would be easier to get across when we could explain the *idea* behind it! The `$throw` argument is straightforward. But why did you do the destructor / `getStatusCode()` thing? What use cases did you have in mind?

And one more: For those 3 methods together (`getHeaders()`, `getContent()` and `toArray()`), do you have some internal (nick) name we could use? Something like "contentFetchers" or so?

Reply:
Ping @nicolas-grekas :-)

Reply:
@nicolas-grekas Could you please merge this, before it ages away? :-)